### ISSUE TITLE
Feature Request: Add fees to transaction and trade entries

### ISSUE THREAD

**Issue by zachgoll** [maybe-finance/maybe#1237]
### Discussed in https://github.com/maybe-finance/maybe/discussions/1235

<div type='discussions-op-text'>

<sup>Originally posted by **marc22polo** October  3, 2024</sup>
### Describe the feature

A lot of banks (at least in my country) charge you a fee for transferring money between your accounts and because Maybe transfers the whole amount it does not show the correct total amount in the end (usually not a big amount, like sub 0.5€ but in the long run it adds up :)). 
Also maybe add an option to choose who the bearer of the fee is (the receiver or the sender).

### Why is this feature important?

The current work around is to create another transaction alongside the transfer transaction for the transfer fee. This would save time for such occasions.

### Additional context, screenshots, and relevant links

_No response_</div>

**Comment by zachgoll** [comment]
@alvaro-rrdt nope, this would just be a simple addition to the schema and the trade forms where users can hard-code in a fee alongside each transaction.

I think this is broad enough where we can put it on the `Account::Entry` model and have it apply to transactions (currency conversion fees), trades (stock purchase fees), and valuations (asset valuation fees from external service).

**Comment by bastervrugt** [comment]
I agree with Zak, but a fee is not always directly related to a transaction. For example, a broker might charge fees for being connected to a specific stock exchange (e.g., DEGIRO). Adding a transaction type 'Fee' could be a good solution, complementing the fee on buy/sell/etc. transactions?

**Comment by zachgoll** [comment]
@bastervrugt yep, we could add a "Fee" type transaction!  Should be very similar to "Interest" that we already have.

**Comment by piru72** [comment]
![Screenshot_2025-01-16-13-19-08-875_com realbyteapps moneymanagerfree-edit](https://github.com/user-attachments/assets/4593c4b5-5efe-49ed-8d43-a66a3e49ca5c)

In my country sometime there are Fee and sometime there aren't. So keeping it optional and only adding it when necessary would do the work. 

Also some banks charges fixes amount of money while some banks charge a certain percentage.

**Comment by sup3rgiu** [comment]
+1

I often transfer money between my primary bank account and a secondary account that I use to buy things online. The only way to do this is by wire transfer, and my primary bank charges a small fixed fee for each wire transfer.

**Comment by tevearitli001** [comment]
Hi, regarding incorporating fees for currency conversion transfers, I am not sure the best way is to add a fee as one could also see the fee as a deterioration of the conversion rate a client actually got for transferring X [ABC] into Y [CBA].Creating a conversion rate = CBA/ABC = Y/abs(X) where X = x - abs(fee[ABD]) (if x is negative as going out from the account)
